src/app.rs

```diff
@@ -707,6 +707,35 @@ impl App {
         }
     }
 
+    pub fn toggle_save_track(&mut self, track_id: String) {
+        if let Some(spotify) = &self.spotify {
+            match spotify.current_user_saved_tracks_contains(&[track_id.clone()]) {
+                Ok(saved) => {
+                    if saved[0] {
+                        match spotify.current_user_saved_tracks_delete(&[track_id]) {
+                            Ok(()) => {}
+                            Err(e) => {
+                                self.handle_error(e);
+                            }
+                        }
+                    } else {
+                        match spotify.current_user_saved_tracks_add(&[track_id]) {
+                            Ok(()) => {}
+                            Err(e) => {
+                                self.handle_error(e);
+                            }
+                        }
+                    }
+                }
+                Err(e) => {
+                    self.handle_error(e);
+                }
+            }
+        };
+    }
+
+    // Currently unused but keep for future changes
+    #[allow(dead_code)]
```

Comment:
Are you able to delete this? No need to allow dead code (shouldn't exist in master).